doc/source/notebooks/gpflow_2_upgrade/gpflow2_upgrade_guide.ipynb

```diff
@@ -0,0 +1,299 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Kernel Input Dims\n",
+    "\n",
+    "The `input_dim` parameter has been removed from the `Kernel` class’s initialiser. Therefore all calls to create a kernel must be changed to remove the `input_dim` parameter.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"files/kernel_input_dims_1_new.png\">\n",
+    "<img src=\"files/kernel_input_dims_2_new.png\">\n",
+    "\n",
+    "**Note**: old code may still run without obvious errors against GPflow, since many kernels take an optional numerical value as their first parameter. You may not get the result you expect though!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Parameter and tf.Variable\n",
+    "\n",
+    "The `Parameter` class in GPflow 1 was a separate class from `tf.Variable`. The `params_as_tensors` decorator or the `params_as_tensors_for` context manager were required to turn them into a tensor that could be consumed by TensorFlow operations. \n",
+    "\n",
+    "In GPflow 2, `Parameter` is a subclass of `tf.Module` that wraps a `tf.Variable`, and can directly be used in place of a tensor, so no such conversion is necessary. \n",
+    "\n",
+    "References to `params_as_tensors` and `params_as_tensors_for` can simply be removed.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Parameter Assignment\n",
+    "\n",
+    "In GPflow 2 the semantics of assigning values to parameters has changed. It is now necessary to use the Parameter.assign method rather than assigning values directly to parameters. For example:\n",
+    "\n",
+    "<img src=\"files/constant.png\">\n",
+    "\n",
+    "In the above example, the old (GPflow 1) code would have assigned the value of `likelihood.scale` to 0.1 (assuming that likelihood is a `Parameterized` object and scale is a `Parameter`), rather than replacing the `scale` attribute with a Python float (which would be the “normal” Python behaviour). This maintains the properties of the parameter. For example, it remains trainable etc.\n",
+    "\n",
+    "In GPflow 2, it is necessary to use the `Parameter.assign` method explicitly to maintain the same behaviour, otherwise the parameter attribute will be replaced by an (untrainable) constant value.\n",
+    "\n",
+    "To change other properties of the parameter (for example, to change transforms etc) you may need to replace the entire parameter object. See [this notebook](../understanding/models.ipynb#Constraints-and-trainable-variables) for further details."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SciPy Optimizer\n",
+    "\n",
+    "Usage of the GPflow’s Scipy optimizer has changed. It has been renamed from `gpflow.train.ScipyOptimizer` to `gpflow.optimizers.Scipy` and its `minimize` method has changed in the following ways:\n",
```

Comment:
Here, I would add that Scipy's minimize method has same configuration arguments as an original scipy's minimize method: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html.

Reply:
I've added a sentence on this.